kernel/src/main/java/org/kframework/compile/checks/CheckSmtLemmas.java

```diff
@@ -0,0 +1,49 @@
+package org.kframework.compile.checks;
+
+import org.kframework.definition.Module;
+import org.kframework.definition.Rule;
+import org.kframework.definition.Sentence;
+import org.kframework.kore.KApply;
+import org.kframework.kore.KRewrite;
+import org.kframework.kore.VisitK;
+import org.kframework.utils.errorsystem.KEMException;
+
+import java.util.Set;
+
+public class CheckSmtLemmas {
+    private final Set<KEMException> errors;
+    private final Module m;
+
+    public CheckSmtLemmas(Set<KEMException> errors, Module m) {
+        this.errors = errors;
+        this.m = m;
+    }
+
+    public void check(Sentence sentence) {
+        if ((sentence instanceof Rule) && sentence.att().contains("smt-lemma")) {
+            check((Rule) sentence);
+        }
+    }
+
+    private void check(Rule rule){
+        if (!(rule.body() instanceof KRewrite)) {
+            return;
+        }
+        new VisitK() {
+            @Override
+            public void apply(KRewrite k) {
```

Comment:
Are you sure you have to `@Override` this? The base class seems to have this code already.
https://github.com/kframework/k/blob/master/kore/src/main/scala/org/kframework/kore/transformers.scala#L95